src/throttle.rs

```diff
@@ -76,6 +76,13 @@ pub struct ThrottleConfig {
     pub wait_put_per_call: Duration,
 }
 
+/// Sleep only if non-zero duration
+async fn sleep(duration: Duration) {
```

Comment:
This is needed because sleep is very low-resolution on Windows, and sleeping for 0ms will end up sleeping for a non-negligible length of time